SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialWorkerConnection.cpp

```diff
@@ -9,8 +9,87 @@
 #include "Async/Async.h"
 #include "Misc/Paths.h"
 
+#if WITH_EDITOR
+#include "Editor.h"
+#include "Settings/LevelEditorPlaySettings.h"
+#endif
+
 DEFINE_LOG_CATEGORY(LogSpatialWorkerConnection);
 
+#if WITH_EDITOR
+struct EditorWorkerController
+{
+	void OnPrePIEEnded(bool bValue)
+	{
+		LastPIEEndTime = FDateTime::Now().ToUnixTimestamp();
+		FEditorDelegates::PrePIEEnded.Remove(PIEEndHandle);
+	}
+
+	void InitWorkers(const FString& WorkerType)
+	{
+		ReplaceProcesses.Empty();
+
+		const int64 WorkerReplaceThreshold = 8;	// Issue replace worker request if restarting deployment within this timeframe
+		int64 SecondsSinceLastSession = FDateTime::Now().ToUnixTimestamp() - LastPIEEndTime;
+		UE_LOG(LogSpatialWorkerConnection, Log, TEXT("Seconds since last session - %d"), SecondsSinceLastSession);
+
+		PIEEndHandle = FEditorDelegates::PrePIEEnded.AddRaw(this, &EditorWorkerController::OnPrePIEEnded);
+
+		int32 PlayNumberOfServers;
+		GetDefault<ULevelEditorPlaySettings>()->GetPlayNumberOfServers(PlayNumberOfServers);
+
+		WorkerIds.SetNum(PlayNumberOfServers);
+		for (int i = 0; i < PlayNumberOfServers; ++i)
+		{
+			FString NewWorkerId = WorkerType + FGuid::NewGuid().ToString();
+
+			if (!WorkerIds[i].IsEmpty() && SecondsSinceLastSession < WorkerReplaceThreshold)
+			{
+				ReplaceProcesses.Add(ReplaceWorker(WorkerIds[i], NewWorkerId));
+			}
+
+			WorkerIds[i] = NewWorkerId;
+		}
+
+	}
+
+	FProcHandle ReplaceWorker(const FString& OldWorker, const FString& NewWorker)
+	{
+		const FString CmdExecutable = TEXT("spatial.exe");
+
+		const FString CmdArgs = FString::Printf(
+			TEXT(" local worker replace "
+				"--existing_worker_id %s "
+				"--local_service_grpc_port 22000 "
```

Comment:
Is it safe to have 22000 port hardcoded here?

Reply:
This is actually the default, and we don't define it anywhere else, so I'm just going to remove it.